includes/abstracts/abstract-wc-rest-terms-controller.php

```diff
@@ -533,6 +533,14 @@ public function delete_item( $request ) {
 		}
 
 		$term = get_term( (int) $request['id'], $taxonomy );
+		// Get default category id.
+		$default_category_id = absint( get_option( 'default_product_cat', 0 ) );
+
+		// Prevent deleting the default product category.
+		if ( $default_category_id === $term ) {
```

Comment:
`$term` will be a term object and `$default_category_id` is an integer, so this comparison won't work. You should be able to compare `$default_category_id` with `(int) $request['id']`, though.